lib/domain/models/memo_collection_metadata.dart

```diff
@@ -0,0 +1,21 @@
+/// Fundamental fields that defines a `Memo` when relating to a `Collection`
+class MemoCollectionMetadata {
```

Comment:
Shouldn't this be abstract?

Reply:
Also, I would like to raise a discussion about this naming. These metadata are related to a memo, and they are used by `Memo` and `MemoExecution` entities correct? If so, why don't call it just `MemoMetadata`?

Reply:
No, they are also instantiated as a list of local memo associated with a local collection file (.json), in `CollectionMemos` transient

Reply:
But this class has no relation with a Collection on its scope. It has an id, a question and an answer. I don't see why having Collection on its name. Even on CollectionMemos class, its responsibility is to hold the memos metadata. Although it can be used in a context that has relation to a Collection, the class scope itself doesn't have any reference to a collection, as far as I can see